### PR TITLE
Refractor

### PR DESCRIPTION
**Pull request by kkmanuu**
✔️ We adhere to YAGNI, KISS, and DRY principles and follow the Ruby style guide meticulously.
✔️ We strictly adhere to SOLID principles.
✔️ We place a strong emphasis on the "Single-responsibility Principle."
✔️ We ensure that the primary role of main.rb is solely to initiate the application and solicit the user's initial option choice or exit.
✔️ We effectively modularize our code by segregating logic into separate classes.

### CODE REVIEW

File: classes/app.rb [hunk 1]
```diff
@@ -37,67 +26,71 @@ def list_people
...
   end
 
   # option 3 - Create a person
   def create_person
-    print 'Do you want to create a student (1) or a teacher (2)? [Input the number]: '
     option = gets.chomp
+    print 'Name: '
+    name = gets.chomp.capitalize
+    print 'Age: '
+    age = gets.to_i
```
Comment (Ibizugbe) [inline classes/app.rb:50]: Nice Job so far👍🏼. In obeying the Single responsibility rule as required, I noticed you have assigned the responsibility of input validation and request to the `main.rb` and as regards the rule, a class should have one responsibility👍🏼. But I have noticed that you are still assigned the responsibility of requesting and validating input in the app class, which defeats the rule😉. Kindly do well to ensure all commands/methods regarding input requests and validation are taken to the `main.rb` file to stay consistent with the single responsibility rule👍🏼.

you can make use of [this lesson](https://www.digitalocean.com/community/conceptual-articles/s-o-l-i-d-the-first-five-principles-of-object-oriented-design) for clarity

**Review by Ibizugbe — CHANGES_REQUESTED** [review]
Hi @Team👋🏼👋🏼, 

require_changes


Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights✅✅✅

- [x] The `main` class has been assigned the responsibility of input validation and request👍🏼.


## Required Changes ♻️

- [x] Kindly take a look at you app and try to ensure all options work properly as required. currently, it seems impossibe to exit the application and there are some issues with adding a rental😉. Kindly do well to check properly and ensure everything works fine👍🏼.
image


_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: classes/app.rb (hunk 1 above)
Comment (kkmanuu) [reply classes/app.rb:50]: @Ibizugbe  Done making changes

**Review by Strangeal — APPROVED** [review]
Hi @kkmanuu, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 




## Highlights
- Single responsibility rule ✅
- Rentals ✅

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**